mvrx/src/main/kotlin/com/airbnb/mvrx/MvCorStateStore.kt

```diff
@@ -0,0 +1,117 @@
+package com.airbnb.mvrx
+
+import io.reactivex.Observable
+import io.reactivex.disposables.Disposable
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.ConflatedBroadcastChannel
+import kotlinx.coroutines.channels.actor
+import kotlinx.coroutines.channels.consumeEach
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.rx2.asObservable
+import java.util.*
+import kotlin.coroutines.CoroutineContext
+
+
+/**
+ * This is a container class around the actual state itself. It has a few optimizations to ensure
+ * safe usage of state.
+ *
+ * All state reducers are run in an actor to ensure that they don't have race
+ * conditions with each other.
+ *
+ */
+
+internal open class MvCorStateStore<S : Any>(initialState: S, final override val coroutineContext: CoroutineContext = Dispatchers.Default) : Disposable, CoroutineScope, MvRxStateStore<S> {
+
+    /**
+     * The stateChannel is where state changes should be pushed to.
+     */
+    private val stateChannel = ConflatedBroadcastChannel(initialState)
```

Comment:
ConflatedBroadcastChannel can drop events. It it expected behavior for MvCorStateStore?

```kotlin
  @Test
  fun test_ConflatedBroadcastChannel_runBlocking() {
    runBlocking {
      val channel = ConflatedBroadcastChannel<Int>()      val collect = async {
        channel.asFlow().collect {
          println("collected $it")
        }
      }
      val send = async {
        (0 until 5).forEach {
          println("send $it")
          channel.send(it)
        }
        channel.cancel()
      }      listOf(collect, send).awaitAll()
    }    
    // console out
    // send 0
    // send 1
    // send 2
    // send 3
    // send 4
    // collected 0
    // collected 4
  }
```

Reply:
@rougsig This PR has been superceded by https://github.com/airbnb/MvRx/pull/347